### PR TITLE
Add Turret

### PR DESCRIPTION
**Pull request by Dzierzan**
Basic implementation of Turret.

### CODE REVIEW

File: mods/hv/weapons/ballistics.yaml [hunk 1]
```diff
@@ -52,3 +52,23 @@ SmallArtillery:
 		Falloff: 100, 55, 20, 5
 		Versus:
 			None: 60
+
+^TurretCannon:
```
Comment (GraionDilach) [inline mods/hv/weapons/ballistics.yaml]: The ^ from the prefix should be dropped.

File: mods/hv/weapons/ballistics.yaml [hunk 2]
```diff
@@ -52,3 +52,23 @@ SmallArtillery:
...
+		Spread: 128
+		Damage: 5000
+		Versus:
+			None: 30
+			Steel: 75
+			Light: 75
+	Warhead@GroundEffect: CreateEffect
+		Explosions: small
+		ValidTargets: Ground
+		ImpactSounds: explosion02.wav
```
Comment (GraionDilach) [inline mods/hv/weapons/ballistics.yaml]: Missing newline.

File: mods/hv/sequences/buildings.yaml [hunk 3]
```diff
@@ -74,3 +74,16 @@ starport:
...
+		Start: 0
+	damaged-idle: turret.png
+		Start: 1
+	turret: turret.png
+		Start: 2
+		Facings: 8
+	shoot: turret.png
+		Start: 10
+		Facings: 8
+	icon: blank.png
```
Comment (GraionDilach) [inline mods/hv/sequences/buildings.yaml]: Missing newline.

**Comment by GraionDilach** [comment]
This needs a rebase.

**Comment by Dzierzan** [comment]
Yes, due lack of my knowledge of github, I feel like I screwed up this PR with interactive rebase. At this point it would better to just close it or cherrypick stuff which is useful.

**Comment by GraionDilach** [comment]
The main issue is in your master branch which you kept merging the updates into instead of constantly rebasing.

If you give me write access to your repo, I can probably fix that, but you then need to restart your local branches as well, which I don't know how feasible would be to you.

File: mods/hv/rules/buildings.yaml [hunk 4]
```diff
@@ -323,14 +323,68 @@ FACTORY2:
 	ProductionBar:
 		ProductionType: Vehicle
 	Power:
-		Amount: -150
 	ProvidesPrerequisite@buildingname:
 	WithDecoration@primary:
 		RequiresSelection: true
+		Amount: -150
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml]: This looks wrong.

File: mods/hv/rules/buildings.yaml [hunk 5]
```diff
@@ -323,14 +323,68 @@ FACTORY2:
...
+	RevealsShroud@Offline:
+		RequiresCondition: !disabled
+		Range: 5c0
+	ProvidesRadar:
+		RequiresCondition: disabled
+	Power:
+		Amount: -60
+		RequiresCondition: !disabled
+	ProvidesPrerequisite@buildingname:
+		Weapon: ^TurretCannon
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml]: This is wrong.

File: mods/hv/rules/buildings.yaml [hunk 6]
```diff
@@ -297,19 +297,19 @@ FACTORY2:
 		Queue: Building
 		BuildPaletteOrder: 90
 		Description: Builds Tanks.
-	Valued:
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml]: This the first error which crashes the Travis build.

File: mods/hv/rules/buildings.yaml [hunk 7]
```diff
@@ -323,14 +323,68 @@ FACTORY2:
...
+
+TURRET:
+	Inherits: ^BaseBuilding
+	Inherits@IDISABLE: ^DisableOnLowPower
+	Inherits@shape: ^2x2Shape
+	Selectable:
+		UseTargetableCellsOffsets: false
+		Bounds: 32,32
+	HitShape:
+		UseTargetableCellsOffsets: false
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml]: > refers to a trait field `UseTargetableCellsOffsets` that does not exist on `Selectable`.

File: mods/hv/rules/buildings.yaml [hunk 8]
```diff
@@ -323,14 +323,68 @@ FACTORY2:
...
+		Cost: 1000
+	Tooltip:
+		Name: Turret
+	Building:
+		Footprint: == xx
+		Dimensions: 2,2
+		LocalCenterOffset: -100,100,0
+	Targetable:
+	Health:
+		TargetTypes: Ground, Structure
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml]: > refers to a trait field `TargetTypes` that does not exist on `Health`.

File: mods/hv/rules/buildings.yaml [hunk 9]
```diff
@@ -323,14 +323,68 @@ FACTORY2:
...
+	ProvidesPrerequisite@buildingname:
+		Weapon: ^TurretCannon
+	Armament:
+		Weapon: TurretCannon
+		LocalOffset: 1200,50,0
+	AttackTurreted:
+	WithSpriteTurret:
+	WithTurretAttackAnimation:
+	Turreted:
+		Sequence: shoot
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml]: > refers to a trait field `Sequence` that does not exist on `Turreted`

File: mods/hv/rules/buildings.yaml [hunk 10]
```diff
@@ -323,14 +323,68 @@ FACTORY2:
...
+	RevealsShroud:
+		Range: 15c0
+	RevealsShroud@Offline:
+		RequiresCondition: !disabled
+		Range: 5c0
+	ProvidesRadar:
+		RequiresCondition: disabled
+	Power:
+		Amount: -60
+		RequiresCondition: !disabled
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml]: This short circuits on low power.

**Comment by Mailaender** [comment]
Added it in https://github.com/Mailaender/OpenHV/commit/b02bd18093935587e901a181ad54122a7ce2414d based upon your rules. Major difference is the defense queue and power down working.

**Pull request closed without merging**